------------------------------------------------------------
revno: 252
committer: Scott Moser <smoser@ubuntu.com>
branch nick: trunk
timestamp: Fri 2010-08-13 11:42:57 -0400
message:
  add support for '%s' in base of seed.
  
  if 'base' input to reed_seeded contains a "%s", then substitute
  'user-data' and 'meta-data' at that location rather than at the end.
  
  Ie:
  - base="http://foo.bar/"
     userdata_url = http://foo.bar/user-data
     metadata_url = http://foo.bar/meta-data
  - base="http://foo.bar/%s?user=smoser"
     userdata_url = http://foo.bar/user-data&user=smoser"
     metadata_url = http://foo.bar/meta-data&user=smoser"
=== modified file 'cloudinit/util.py'
--- old/cloudinit/util.py	2010-08-12 16:00:23 +0000
+++ new/cloudinit/util.py	2010-08-13 15:42:57 +0000
@@ -142,8 +142,12 @@
     if base.startswith("/"):
         base="file://%s" % base
 
-    ud_url = "%s%s%s" % (base, "user-data", ext)
-    md_url = "%s%s%s" % (base, "meta-data", ext)
+    if base.find("%s") >= 0:
+        ud_url = base % ("user-data" + ext)
+        md_url = base % ("meta-data" + ext)
+    else
+        ud_url = "%s%s%s" % (base, "user-data", ext)
+        md_url = "%s%s%s" % (base, "meta-data", ext)
 
     try:
         md_resp = urllib2.urlopen(urllib2.Request(md_url), timeout=timeout)

